src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/DraftApprenticeshipController.cs

```diff
@@ -110,7 +116,53 @@ public async Task<IActionResult> AddDraftApprenticeship(AddDraftApprenticeshipVi
             }
         }
 
-        private async Task AddLegalEntityAndCoursesToModel(AddDraftApprenticeshipViewModel model)
+        [HttpGet]
+        [Route("edit-apprentice/{DraftApprenticeshipHashedId}")]
+        public async Task<IActionResult> EditDraftApprenticeship(EditDraftApprenticeshipRequest request)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var imodel = await _providerCommitmentsService.GetDraftApprenticeshipForCohort(request.CohortId.Value, request.DraftApprenticeshipId.Value);
+            var model = _editDraftApprenticeshipDetailsToViewModelMapper.Map(imodel);
+
+            await AddLegalEntityAndCoursesToModel(model);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("edit-apprentice/{DraftApprenticeshipHashedId}")]
+        public async Task<IActionResult> EditDraftApprenticeship(EditDraftApprenticeshipViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                await AddLegalEntityAndCoursesToModel(model);
+                return View(model);
+            }
+
+            var updateRequest = _updateDraftApprenticeshipRequestMapper.Map(model);
+            // TODO Check if UserId is needed here
```

Comment:
Are we now in a position to resolve this todo?

Reply:
Just needed to know if the UserId was needed for backward compatibility. The existing system seems to use a LastUpdated structure to record who updated the apprenticeship. We currently don't do this within the V2 update